run-scripts/run-custom-dataset.py

```diff
@@ -0,0 +1,263 @@
+        "--host", db_config['host'],
+        "--db-name", db_config['db-name']
+    ]
+
+    # Handle initial flags (no skip for the first ef_search)
+    if case.get("db-name", True):
+        base_command.append("--drop-old")
+    else:
+        base_command.append("--skip-drop-old")
+
+    if case.get("load", True):
+        base_command.append("--load")
+    else:
+        base_command.append("--skip-load")
+
+    if case.get("search-serial", True):
+        base_command.append("--search-serial")
+    else:
+        base_command.append("--skip-search-serial")
+
+    if case.get("search-concurrent", True):
+        base_command.append("--search-concurrent")
+    else:
+        base_command.append("--skip-search-concurrent")
+
+    if case.get("custom-dataset-use-shuffled", True):
+        base_command.append("--custom-dataset-use-shuffled")
+    else:
+        base_command.append("--skip-custom-dataset-use-shuffled")
+
+    base_command.extend([
+        "--case-type", case["case-type"],
+        "--custom-case-name", str(case["custom-case-name"]),
+        "--custom-dataset-name", str(case["custom-dataset-name"]),
+        "--custom-dataset-dir", str(case["custom-dataset-dir"]),
+        "--custom-dataset-size", str(case["custom-dataset-size"]),
+        "--custom-dataset-dim", str(case["custom-dataset-dim"]),
+        "--custom-dataset-file-count", str(case["custom-dataset-file-count"]),
+        "--maintenance-work-mem", case["maintenance-work-mem"],
+        "--max-parallel-workers", str(case["max-parallel-workers"]),
+        "--ef-construction", str(case["ef-construction"]),
+        "--m", str(case["m"]),
+        "--k", str(case["k"]),
+        "--num-concurrency", case["num-concurrency"],
+    ])
+
+    run_count = case.get("run-count", 1)  # Default to 1 if not specified
+
+
+    for run in range(run_count):
+        print(f"Starting run {run + 1} of {run_count} for case: {case['db-label']}")
+        for i, ef_search in enumerate(case["ef-search"]):
+            command = base_command + ["--ef-search", str(ef_search)]
+
+            # Build the index only once.
+            if i > 0 or run > 0:
+                # Remove conflicting --drop-old and --load flags
+                command = [arg for arg in command if arg not in ["--drop-old", "--load"]]
+                # Add skip flags if they are not already in the command
+                if "--skip-drop-old" not in command:
+                    command.append("--skip-drop-old")
+                if "--skip-load" not in command:
+                    command.append("--skip-load")
+
+            try:
+                random_number = random.randint(1, 100000)
+                print(f"Running command: {' '.join(command)}")
+                output_dir = f"results/pgvector/hnsw/{case['db-label']}/{db_config['provider']}/{db_config['instance-type']}-{str(case['m'])}-{str(case['ef-construction'])}-{ef_search}-{case['case-type']}-{run}-{random_number}"
```

Comment:
@Sheharyar570 output dir has `pgvector/hnsw` hardcorded. I guess this was not the final version. Also i'm thinking we can skip adding this file in this PR, assuming this file is for constraint mem case?